src/Core/Context.idr

```diff
@@ -1628,6 +1628,14 @@ hasFlag fc n fl
               | Nothing => undefinedName fc n
          pure (fl `elem` flags def)
 
+export
+hasSetTotal : {auto c : Ref Ctxt Defs} -> FC -> Name -> Core Bool
```

Comment:
Why is this needed in addition to `hasFlag` (above)?

Reply:
Because totality can be set by three different flags (`SetTotal Total`, `SetTotal CoveringOnly`, `SetTotal PartialOK`). It is possible to implement this check with `hasFlag`, but 3 calls will be required.